### PR TITLE
Component-based normalisation clean up

### PR DESCRIPTION
**Pull request by KrisThielemans**
move code from the `find_ML_normfactors3D` and `apply_normfactors3D` to new functions/classes, adding a bit more documentation along the way.

### CODE REVIEW

**Comment by KrisThielemans** [comment]
@markus-jehl @emikhay this is one more step along the way to clean up the norm code. Just moving things to different places. @danieldeidda is working on making blocks-on-cylindrical data compatible (needs this PR)

**Comment by markus-jehl** [comment]
The code changes look sensible to me, but it turns out this is not actually straight-forward to test for me, because I completely disentangled that function for our use (i.e. removed the iterations and all of the file output etc.). I don't have any code that can easily call these structures with known expected results.

What I didn't see just from looking at the code: where does BinNormalisationPETFromComponents actually use the geometric factors, efficiencies and blocks to modify the internal projdata object?

**Comment by KrisThielemans** [comment]
> What I didn't see just from looking at the code: where does BinNormalisationPETFromComponents actually use the geometric factors, efficiencies and blocks to modify the internal projdata object?

This sits in its `create_projdata` function. (Just clarifying, there is no code in that class that actually fills the efficiencies etc., only on how to use it).

**Review by danieldeidda — COMMENTED** [review]
looks OK, does it need a parfile example?

File: src/recon_buildblock/BinNormalisationPETFromComponents.cxx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+#include "stir/ViewSegmentNumbers.h"
+#include "stir/Succeeded.h"
+#include "stir/warning.h"
+#include "stir/error.h"
+#include "stir/numerics/divide.h"
+#include <boost/format.hpp>
+
+START_NAMESPACE_STIR
+
+const char* const BinNormalisationPETFromComponents::registered_name = "PETComponents";
```
[the remark this answers is not in the document]
Comment (KrisThielemans) [reply src/recon_buildblock/BinNormalisationPETFromComponents.cxx:34]: no. This class currently cannot parse values and is intended as a base-class for others that use the same functionality. That's more work and I'll probably put that in a derived class (in a future life).

**Pull request merged**